contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,730 @@
+                yieldTokenBalance
+            );
+        }
+    }
+
+    /**
+     * @notice Withdraws stETH from vault
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     */
+    function withdrawYieldToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount
+    ) internal returns (uint256 yieldTokenBalance) {
+        IERC20 collateral = IERC20(collateralToken);
+
+        yieldTokenBalance = collateral.balanceOf(address(this));
+        uint256 yieldTokensToWithdraw =
+            dsmin(yieldTokenBalance, withdrawAmount);
+        if (yieldTokensToWithdraw > 0) {
+            collateral.safeTransfer(recipient, yieldTokensToWithdraw);
+        }
+    }
+
+    /**
+     * @notice Withdraws `asset` from vault
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     * @param yieldTokenBalance is the collateral token (stETH) balance of the vault
+     */
+    function withdrawBaseToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount,
+        uint256 yieldTokenBalance
+    ) internal {
+        uint256 underlyingTokensToWithdraw =
+            IWSTETH(collateralToken).getStETHByWstETH(
+                withdrawAmount.sub(yieldTokenBalance)
+            );
+
+        transferAsset(payable(recipient), underlyingTokensToWithdraw);
+    }
+
+    /**
+     * @notice Unwraps the necessary amount of the yield-bearing yearn token
+     *         and transfers amount to vault
+     * @param amount is the amount of `asset` to withdraw
+     * @param collateralToken is the address of the collateral token
+     * @param crvPool is the address of the steth <-> eth pool on curve
+     * @param crvPoolSlippage is the slippage for the steth <-> eth pool on curve
+     */
+    function unwrapYieldToken(
+        uint256 amount,
+        address collateralToken,
+        address crvPool,
+        uint256 crvPoolSlippage
+    ) external {
+        uint256 assetBalance = address(this).balance;
+
+        uint256 amountToUnwrap =
+            IWSTETH(collateralToken).getWstETHByStETH(
+                dsmax(assetBalance, amount).sub(assetBalance)
+            );
+
+        if (amountToUnwrap > 0) {
+            IWSTETH wsteth = IWSTETH(collateralToken);
+            // Unrap to stETH
+            wsteth.unwrap(amountToUnwrap);
+
+            ICRV crv = ICRV(crvPool);
+
+            // CRV SWAP HERE from steth -> eth
+            // 0 = ETH, 1 = STETH
+            uint256 minETHOut =
+                crv.get_dy(1, 0, amountToUnwrap).mul(
+                    uint256(256).sub(crvPoolSlippage).div(100)
+                );
+
+            crv.exchange(1, 0, amountToUnwrap, minETHOut);
```

Comment:
This approach makes it vulnerable to sandwich attacks. To fix this, we need to pass in the `minETHOut` as a parameter for withdrawals

Reply:
can you elaborate? can't a bot still frontrun this tx even if we pass in some arbitrary minETHOut calculated off-chain?